dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/docker/CAContainer.java

```diff
@@ -0,0 +1,372 @@
+	 * @throws IOException
+	 */
+	public void clearAAAARecord(String host) throws IOException {
+		final String METHOD_NAME = "clearAAAARecord(String)";
+
+		try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+			/*
+			 * Generate the JSON request.
+			 */
+			String jsonString = "{\"host\":\"" + host + "\"}";
+			StringEntity requestEntity = new StringEntity(jsonString, ContentType.APPLICATION_JSON);
+
+			/*
+			 * Create a POST request to the mock DNS server.
+			 */
+			HttpPost httpPost = new HttpPost(getManagementAddress() + "/clear-aaaa");
+			httpPost.setEntity(requestEntity);
+
+			/*
+			 * Send the POST request and process the response.
+			 */
+			try (final CloseableHttpResponse response = httpclient.execute(httpPost)) {
+				AcmeFatUtils.logHttpResponse(ChalltestsrvContainer.class, METHOD_NAME, httpPost, response);
+
+				StatusLine statusLine = response.getStatusLine();
+				if (statusLine.getStatusCode() != 200) {
+					throw new IOException(
+							METHOD_NAME + ": Expected response 200, but received response: " + statusLine);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Clear a previously added A record that was added to the mock DNS server.
+	 * 
+	 * @param host
+	 *            The host / domain to redirect requests for.
+	 * @throws IOException
+	 */
+	public void clearARecord(String host) throws IOException {
+		final String METHOD_NAME = "clearMockARecord(String)";
+
+		try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+			/*
+			 * Generate the JSON request.
+			 */
+			String jsonString = "{\"host\":\"" + host + "\"}";
+			StringEntity requestEntity = new StringEntity(jsonString, ContentType.APPLICATION_JSON);
+
+			/*
+			 * Create a POST request to the mock DNS server.
+			 */
+			HttpPost httpPost = new HttpPost(getManagementAddress() + "/clear-a");
+			httpPost.setEntity(requestEntity);
+
+			/*
+			 * Send the POST request and process the response.
+			 */
+			try (final CloseableHttpResponse response = httpclient.execute(httpPost)) {
+				AcmeFatUtils.logHttpResponse(ChalltestsrvContainer.class, METHOD_NAME, httpPost, response);
+
+				StatusLine statusLine = response.getStatusLine();
+				if (statusLine.getStatusCode() != 200) {
+					throw new IOException(
+							METHOD_NAME + ": Expected response 200, but received response: " + statusLine);
+				}
+			}
+		}
+	}
+	public String getManagementAddress() {
+		return "http://" + this.getContainerIpAddress() + ":" + this.getMappedPort(MANAGEMENT_PORT);
+	}
+
+	public String getAcmeDirectoryURI() {
+		return null;
+	}
+
+	public String getIntraContainerIP() {
+		String intraContainerIpAddress = null;
+		for (Entry<String, ContainerNetwork> entry : getContainerInfo().getNetworkSettings().getNetworks().entrySet()) {
+			intraContainerIpAddress = entry.getValue().getIpAddress();
+			break;
+		}
+		if (intraContainerIpAddress == null) {
+			fail("Didn't find IP address for challtestsrv server.");
```

Comment:
Don't use fail. Throw an IllegalStateException. We don't want to assume everyone is using these in JUnits.